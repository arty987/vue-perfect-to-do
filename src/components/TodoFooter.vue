<script>
export default {
  name: "TodoFooter"
}
</script>

<template>
  <li>
    <span>5 items left</span>
    <button type="button">All</button>
    <button type="button">Active</button>
    <button type="button">Completed</button>
    <button type="button">Clear completed</button>
  </li>
</template>

<style scoped>

</style>