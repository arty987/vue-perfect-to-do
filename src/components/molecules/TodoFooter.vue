<script>
import {mapActions, mapGetters} from "vuex";
import AppButton from "@/components/atoms/AppButton.vue";

export default {
  name: "TodoFooter",
  components: {
    AppButton
  },
  methods: {
    ...mapActions([
      "displayAll",
      "displayCompleted",
      "displayActive",
      "clearCompleted",
    ])
  },
  computed: {
    ...mapGetters([
      "activeCount"
    ])
  }
}
</script>

<template>
  <li>
    <span>{{ activeCount }} items left</span>
    <AppButton @click="displayAll">All</AppButton>
    <AppButton @click="displayActive">Active</AppButton>
    <AppButton @click="displayCompleted">Completed</AppButton>
    <AppButton @click="clearCompleted">Clear completed</AppButton>
  </li>
</template>

<style scoped>

</style>