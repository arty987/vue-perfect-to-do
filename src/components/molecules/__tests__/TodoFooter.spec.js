import {shallowMount} from '@vue/test-utils'
import TodoFooter from '@/components/molecules/TodoFooter.vue';
import {createStore} from 'vuex';
import {fireEvent, render, screen} from '@testing-library/vue';

describe('TodoFooter.vue', () => {

    const getters = {
        activeCount: () => 3,
        filter: () => 'default'
    };

    let actions;
    let store;

    beforeEach(() => {
        actions = {
            setFilter: jest.fn(),
            clearCompleted: jest.fn()
        };

        store = createStore({
            getters,
            actions
        });
    })

    it('Should render TodoFooter', () => {
        const wrapper = shallowMount(TodoFooter, {
            global: {
                plugins: [store]
            }
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it('Should click Clear completed', async () => {
        render(TodoFooter, {
            global: {
                plugins: [store]
            },
        });

        const button = screen.getByText('Clear completed');

        await fireEvent.click(button);

        expect(actions.clearCompleted).toHaveBeenCalled();
    });

    it.each([
        'All',
        'Active',
        'Completed'
    ])('Should click filter buttons', async (filter) => {
        render(TodoFooter, {
            global: {
                plugins: [store]
            },
        });

        const button = screen.getByText(filter);

        await fireEvent.click(button);

        expect(actions.setFilter).toHaveBeenCalledWith(expect.any(Object), filter);
    });
})