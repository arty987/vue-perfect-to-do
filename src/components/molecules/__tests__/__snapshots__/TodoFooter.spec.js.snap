// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoFooter.vue Should render TodoFooter 1`] = `
<li
  class="list-group-item d-flex align-items-center justify-content-between"
>
  <span>
    3 items left
  </span>
  <div>
    
    <app-button-stub
      class=""
    />
    <app-button-stub
      class=""
    />
    <app-button-stub
      class=""
    />
    
  </div>
  <app-button-stub />
</li>
`;
